packages/client/src/pages/userInfoPage/UserInfoPage.tsx

```diff
@@ -1,79 +1,141 @@
-/* eslint-disable react/jsx-props-no-spreading */
 import { Link } from 'react-router-dom';
-import React, { useCallback } from 'react';
-import { Typography, Form, Input, Button, Avatar, Radio } from 'antd';
+import React, { useCallback, useState } from 'react';
+import { Typography, Form, Input, Button, Avatar, Modal, Row, Col } from 'antd';
+import { Controller, useForm } from 'react-hook-form';
+import { yupResolver } from '@hookform/resolvers/yup';
 import { appRoutes } from '@/utils/router/appRoutes';
+import { changeAvatar } from '@/controllers/changeAvatar';
+import { UserProfileForm } from './typings';
 import './UserInfoPage.scss';
+import { profileSchema } from '@/utils/validation/validationSchema';
 
-const layout = {
-  labelCol: { span: 10 },
-  wrapperCol: { span: 34 },
-};
 const { Title } = Typography;
 
-// временная заглушка
-const userProfile = {
-  id: 111,
-  first_name: 'Test',
-  second_name: 'Testov',
-  display_name: 'test',
-  login: 'test111',
-  avatar: '',
-  email: 'test3@test.com',
-  phone: '89998887766',
-};
-
 export const UserInfoPage = () => {
-  // eslint-disable-next-line @typescript-eslint/no-empty-function
-  const submitHandler = useCallback((): void => {}, []);
+  const {
+    control,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<UserProfileForm>({
+    defaultValues: {
+      first_name: 'Test',
+      second_name: 'Testov',
+      display_name: 'test',
+      login: 'test111',
+      avatar: '',
+      email: 'test3@test.com',
+      phone: '89998887766',
+    },
+    mode: 'onChange',
+    resolver: yupResolver(profileSchema),
+  });
+
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [image, setImage] = useState<File>();
+
+  const onSubmit = (data: UserProfileForm) => {
+    // eslint-disable-next-line no-console
+    console.log(data);
+  };
+
+  const handleChangeAvatar = useCallback((event: React.ChangeEvent<HTMLInputElement>): void => {
+    const { files } = event.target;
+    if (!files) {
+      return;
+    }
+    setImage(files[0]);
+  }, []);
+
+  const uploadAvatar = useCallback((): void => {
+    if (image) {
+      const formData = new FormData();
+      formData.append('avatar', image);
+      changeAvatar(formData);
+      setIsModalOpen(false);
+    }
```

Comment:
Чуть лучше читаемость, если использовать early return:
```suggestion
    if (!image) {
      return
    }
    const formData = new FormData();
    formData.append('avatar', image);
    changeAvatar(formData);
    setIsModalOpen(false);
```

Reply:
Довольно субъективно, так что на твое усмотрение )

Reply:
Исправила